x-pack/plugins/enterprise_search/server/lib/indices/pipelines/ml_inference/update_ml_inference_mappings.ts

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ElasticsearchClient } from '@kbn/core-elasticsearch-server';
+
+import { FieldMapping } from '../../../../../common/ml_inference_pipeline';
+
+import { ErrorCode } from '../../../../../common/types/error_codes';
+
+import { isIllegalArgumentException } from '../../../../utils/identify_exceptions';
+
+/**
+ * Creates Elasticsearch field mappings so that the outputs of ML Inference pipelines are not indexed as `float` fields
+ * @param indexName - the index whose mapping will be updated
+ * @param fieldMappings - the array of objects representing the source field (text) names and target fields (ML output) names
+ * @param esClient
+ */
+export const updateMlInferenceMappings = async (
+  indexName: string,
+  fieldMappings: FieldMapping[],
+  esClient: ElasticsearchClient
+) => {
+  const sourceFields = fieldMappings.map(({ sourceField }) => sourceField);
+
+  const nonDefaultTargetFields = fieldMappings
+    .filter(
+      (fieldMapping) =>
+        fieldMapping.targetField !== `ml.inference.${fieldMapping.sourceField}_expanded`
+    )
+    .map((fieldMapping) => fieldMapping.targetField);
+
+  // Today, we only update mappings for text_expansion fields.
+  const mapping = generateTextExpansionMappingProperties(sourceFields, nonDefaultTargetFields);
```

Comment:
Should we combine this logic parsing `fieldMappings` to `mapping` and unit test it? 

Not sure if we are concerned about any possible edge cases where a `fieldMappings` value could result in `mapping` with no source or target fields. I don't think it's currently possible given the usage of the endpoint but looking at the code doing a filter it strikes me as technical possibility. Not sure if its worth being concerned about or not though.

Reply:
And this is why I shouldn't enable auto-merge. :( Sorry Rodney, to have effectively ignored this comment.
If there are no fieldMappings, then no `properties` with a `type` would be defined, so the putMapping request would effectively no-op. I'm not too worried about that edge case.

There are unit tests to validate that these mappings are created correctly, though: https://github.com/elastic/kibana/pull/154291/files/2833e24536e78a7701d531435ea12be1a475984e#diff-b698a365727af822707e5c4ddea43e134500def62ea5e86167b0c59c27c9430fR23-R52